### DISCUSSION TITLE
FillStyleExtension with polygon fill colour

### DISCUSSION THREAD

**Discussion by juneidy** [visgl/deck.gl#6347] [Q&A] [answered]
Hi there, I have managed to get polygon pattern working by using [FillStyleExtension](https://deck.gl/docs/api-reference/extensions/fill-style-extension).

However, I am trying to apply PNG pattern, and I want the transparent part of the PNG to show polygon's fill colour. Is that possible?

The extension seems to treat `getFillColor` as the pattern's colour and no indication of allowing two sets of colour (i.e. polygon-fill-colour and pattern-colour)

**Comment by kevduong1** [comment]
would also like this feature

**Comment by felixpalmer** [comment, marked as answer]
Implemented in https://github.com/visgl/deck.gl/pull/10618
